erts/emulator/beam/erl_bif_persistent.c

```diff
@@ -359,7 +360,7 @@ BIF_RETTYPE persistent_term_put_2(BIF_ALIST_2)
     ctx->key = BIF_ARG_1;
     ctx->term = BIF_ARG_2;
 
-    ctx->entry_index = lookup(ctx->hash_table, ctx->key);
+    ctx->entry_index = lookup(ctx->hash_table, ctx->key, &ctx->old_bucket);
```

Comment:
The first `get_bucket()` call in `persistent_term_put_2` can be replaced with `ctx->old_bucket`:

`if (is_nil(get_bucket(ctx->hash_table, ctx->entry_index))) {
`

Reply:
Good catch, thanks!